backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxSparkPlanExecApi.scala

```diff
@@ -748,6 +749,14 @@ class VeloxSparkPlanExecApi extends SparkPlanExecApi {
     SparkShimLoader.getSparkShims.getExtendedColumnarPostRules() ::: List()
   }
 
+  override def genInjectPostHocResolutionRules(): List[SparkSession => Rule[LogicalPlan]] = {
+    val buf: ListBuffer[SparkSession => Rule[LogicalPlan]] = ListBuffer()
+    if (BackendsApiManager.getSettings.enableNativeArrowReadFiles()) {
+      buf += ArrowConvertorRule
```

Comment:
Would you like to explain a bit about the reason to inject the rule as PostHocResolutionRule? Thanks.

Reply:
Because it replace the LogicalRelation( HadoopFsRelation(_, CSVFileFormat))